### ISSUE TITLE
CalendarIntervalSchedule WithIntervalInWeeks ignores StartAt date

### ISSUE THREAD

**Issue by astynax777** [quartznet/quartznet#1035]
As part of vetting out using Quartz.net for my project, I found an unexpected result when attempting to test Every Week and Every N Weeks.  In my TestMethod below, I set the start date as yesterday and I expect it to be fired again six days from today.  When I call `ITrigger.GetNextFireTimeUtc` it returns the start date of yesterday instead of a week from the start date (since we're past it).  I thought perhaps if I actually ran the scheduler, it would validate the `GetNextFireTimeUtc ` and realize at runtime that it's in the past and not execute the job, but instead it executed the job immediately; therefore I assume it will execute again 7 days from today rather than 7 days from the supplied `StartAt` date.

Here's why this matters to me: 
I want to create an application that stores these scheduled jobs in a database and allow the user to schedule reports on a recurrence (hourly, daily, weekly, bi-weekly, monthly, etc).  When my application runs, it'll pull and monitor jobs in the database and add them to or remove them from the scheduler.  Suppose my application shuts down unexpectedly and it's not discovered for a day or two.  Starting it up again two days later would throw off the day of the week of every weekly, bi-weekly, etc job it loads.  I would expect it to load the schedule information, including the start date, and calculate the first occurrence from that start date.  I initially explored trying to accomplish this via the cron scheduler but it appears there's no way to do weekly intervals on there.

```
private async Task<DateTimeOffset?> GetNextFireTime(IScheduler scheduler, string jobName, string groupName)
{
    var jobKey = new JobKey(jobName, groupName);
    if (await scheduler.CheckExists(jobKey))
    {
        var detail = await scheduler.GetJobDetail(jobKey);
        var triggers = await scheduler.GetTriggersOfJob(jobKey);

        if (triggers.Count > 0)
        {
            var nextFireTime = triggers.First().GetNextFireTimeUtc();
            return nextFireTime;
        }
    }

    return null;
}

[TestMethod]
public async Task TestEveryWeek()
{

    var jobName = "EveryWeek";
    var groupName = "Weekly";
    var factory = new StdSchedulerFactory();
    var scheduler = await factory.GetScheduler();

    var job = JobBuilder.Create<MyJob>()
        .WithIdentity(jobName, groupName)
        .Build();

    // Set start date as yesterday 12PM
    var startAtDateTime = new DateTimeOffset(DateTime.Now.Date.AddHours(12).AddDays(-1));

    var trigger = TriggerBuilder.Create()
        .StartAt(startAtDateTime)
        .WithIdentity(jobName, groupName)
        .WithCalendarIntervalSchedule(x => x.WithIntervalInWeeks(3))
        .Build();

    await scheduler.ScheduleJob(job, trigger);

    // Expect the next time will be 6 days from today at 12PM
    var expectedDate = new DateTimeOffset(DateTime.Now.Date.AddHours(12).AddDays(6));
    var nextDate = await GetNextFireTime(scheduler, jobName, groupName);

    Assert.IsNotNull(nextDate);
    Assert.AreEqual(expectedDate, nextDate);
}
```

Here's the result of my test method:
>  Assert.AreEqual failed. Expected:<11/23/2020 12:00:00 PM -05:00>. Actual:<11/16/2020 12:00:00 PM -05:00>.

If this is not the way to accomplish this and there is an alternative, please let me know.

**Comment by astynax777** [comment]
I don't know why I didn't see this before, but changing my search keywords found an almost identical issue/use case as mine here:
#636 
StartAt functions as expected with past dates on cron triggers, so I would argue it should function identically here.  I saw the suggestion regarding using the AdoJobStore.  The disclaimer about it being slower scared me away from this approach.  I'll look into this as well as a potential workaround.

**Comment by astynax777** [comment]
Following up on this, I found another reason why the StartAt logic is more important.  I tested using the AdoJobStore.  The AdoJobStore doesn't solve this issue.  If the job is missed, the job runs immediately after the service runs again rather than on the next interval.  Maybe for some this is desired.  For me, however, this throws off the entire interval.  So, if I were to have a bi-weekly report that runs on Monday, and I found out on Tuesday that the service stopped, then starting it again on Tuesday would run it immediately and cause it to trigger every two weeks on Tuesday.  I would have to remove the existing trigger and setup a new one to run again the Monday after next.  If the "StartAt" logic was respected and used in scheduling, then it would look at the last run time or StartAt date and calculate the next fire time based on the recurrence interval.  For those of us who think of scheduling along the lines of MS Office recurrences, we create a schedule and don't expect the schedule to shift.  That's how my users will think when they schedule reports.  Perhaps having an option to choose whether or not to "run immediately if it's missed" would be nice.
